babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -236,6 +236,39 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
     { return new SqlCreateAttributeJournalTable(id, getPos()); }
 }
 
+SqlCreateAttribute CreateTableAttributeJournal() :
+{
+  JournalType journalType = JournalType.UNSPECIFIED;
+  JournalModifier journalModifier = JournalModifier.UNSPECIFIED;
+}
+{
+    (
+        (
+            <LOCAL> { journalModifier = JournalModifier.LOCAL; }
+        |
+            <NOT> <LOCAL> { journalModifier = JournalModifier.NOT_LOCAL; }
+        )
+        <AFTER> <JOURNAL> { journalType = JournalType.AFTER; }
+    |
+        [
```

Comment:
This is looking a lot clearer already - thanks!

To decrease the amount of nesting even further, suggest removing the outer brackets and instead having an "else" case for the <NO>/<DUAL> and <BEFORE>/<AFTER> clauses below. Where in each "else" case, JournalType.UNSPECIFIED or JounralModifier.UNSPECIFIED are explicitly set.

Reply:
Ah I see, that would look a lot cleaner. I'll do that and remove the UNSPECIFIED from the above